superset/assets/spec/javascripts/explore/utils_spec.jsx

```diff
@@ -12,48 +13,124 @@ describe('utils', () => {
+        URI(url),
         URI('/superset/explore_json/table/1/')
-          .search({ form_data: sFormData, force: 'true' }),
+          .search({ force: 'true' }),
     );
+    expect(payload).to.deep.equals(formData);
   });
-  it('getExploreUrl generates proper csv URL', () => {
+  it('generates proper csv URL', () => {
+    const { url, payload } = getExploreUrlAndPayload({
+      formData,
+      endpointType: 'csv',
+      force: false,
+      curUrl: 'superset.com',
+    });
     compareURI(
-        URI(getExploreUrl(formData, 'csv', false, 'superset.com')),
+        URI(url),
         URI('/superset/explore_json/table/1/')
-          .search({ form_data: sFormData, csv: 'true' }),
+          .search({ csv: 'true' }),
     );
+    expect(payload).to.deep.equals(formData);
   });
-  it('getExploreUrl generates proper standalone URL', () => {
+  it('generates proper standalone URL', () => {
+    const { url, payload } = getExploreUrlAndPayload({
+      formData,
+      endpointType: 'standalone',
+      force: false,
+      curUrl: 'superset.com',
+    });
     compareURI(
-        URI(getExploreUrl(formData, 'standalone', false, 'superset.com')),
+        URI(url),
         URI('/superset/explore/table/1/')
-          .search({ form_data: sFormData, standalone: 'true' }),
+          .search({ standalone: 'true' }),
     );
+    expect(payload).to.deep.equals(formData);
   });
-  it('getExploreUrl preserves main URLs params', () => {
+  it('preserves main URLs params', () => {
+    const { url, payload } = getExploreUrlAndPayload({
+      formData,
+      endpointType: 'json',
+      force: false,
+      curUrl: 'superset.com?foo=bar',
+    });
     compareURI(
-        URI(getExploreUrl(formData, 'json', false, 'superset.com?foo=bar')),
+        URI(url),
         URI('/superset/explore_json/table/1/')
-          .search({ foo: 'bar', form_data: sFormData }),
+          .search({ foo: 'bar' }),
+    );
+    expect(payload).to.deep.equals(formData);
+  });
+  it('generate proper save slice url', () => {
+    const { url, payload } = getExploreUrlAndPayload({
+      formData,
+      endpointType: 'json',
+      force: false,
+      curUrl: 'superset.com?foo=bar',
+    });
+    compareURI(
+      URI(url),
+      URI('/superset/explore_json/table/1/')
+        .search({ foo: 'bar' }),
+    );
+    expect(payload).to.deep.equals(formData);
+  });
+  it('generate proper saveas slice url', () => {
+    const { url, payload } = getExploreUrlAndPayload({
+      formData,
+      endpointType: 'json',
+      force: false,
+      curUrl: 'superset.com?foo=bar',
+    });
+    compareURI(
+      URI(url),
+      URI('/superset/explore_json/table/1/')
+        .search({ foo: 'bar' }),
+    );
+    expect(payload).to.deep.equals(formData);
+  });
+  it('getExploreLongUrl generates proper base url with form_data', () => {
+    compareURI(
+      URI(getExploreLongUrl(formData, 'base', false, 'http://superset.com')),
```

Comment:
are the last two arguments used by the `getExploreLongUrl ` func? I think functions that take an object are much less error-prone than relying on positional arguments.

Reply:
last 2 arguments are not used! fixed.